src/utils/libfabric/libfabric_topology.cpp

```
@@ -195,6 +195,48 @@ nixlLibfabricTopology::isValidDevice(const std::string &efa_device) const {
+    if (ret) {
+        NIXL_ERROR << "fi_getinfo failed for PCIe mapping with provider " << provider_name << ": "
+                   << fi_strerror(-ret);
+        fi_freeinfo(hints);
+        return NIXL_ERR_BACKEND;
+    }
+
+    device_found = false;
+    for (struct fi_info *cur = info; cur; cur = cur->next) {
+        if (cur->domain_attr && cur->domain_attr->name && cur->nic && cur->nic->bus_attr) {
+            // this is enough, we have bus_attr
+            device_found = true;
```

Cppcheck (2.20.0)

[style] The function 'isTopologyDiscovered' is never used.

(unusedFunction)